mesa: Use the llvm-config of spec['llvm'] for '+llvm'

bernhardkaindl opened the pull request:
When running `spack install mesa@@21.2.3 %clang@8.0.1` on a host which has `/usr/bin/llvm-config-13`, meson picked /usr/bin/llvm-config-13, which in turn caused it to fail to build since it is not yet updated for llvm-13.

Therefore:

Fix builds on hosts where /usr/bin/llvm-config-* is found and provides an incompatible version:

Ensure that the `llvm-config` of `spec['llvm'].prefix` is used.

alalazo reviewed (changes requested):
There seem to be a legitimate failure in the pipeline: https://gitlab.spack.io/spack/spack/-/jobs/1210117

bernhardkaindl committed:
mesa: Use the llvm-config of spec['llvm'] for '+llvm'

Fix builds on hosts where /usr/bin/llvm-config-* is found and provides an
incompatible version: Ensure that the llvm-config of spec['llvm'] is used.

bernhardkaindl commented:
> There seem to be a legitimate failure in the pipeline: https://gitlab.spack.io/spack/spack/-/jobs/1210117

Thanks! Fixed now.
